app/models/controlled_work_document_content.rb

```diff
@@ -0,0 +1,206 @@
+# in the session, which contains all the information CFE returned when we asked it to perform an
+# eligibility calculation for this check
+class ControlledWorkDocumentContent < Check
+  include ActionView::Helpers::NumberHelper
+
+  def from_cfe_payload(path)
+    path_parts = path.split(".").map { _1.to_i.to_s == _1 ? _1.to_i : _1 }
+    value = session_data.dig("api_response", *path_parts)
+    format(value)
+  end
+
+  def from_attribute(attribute)
+    format(send(attribute))
+  end
+
+private
+
+  def format(value)
+    return unless value
+    return value unless value.is_a?(Numeric)
+
+    precision = value.round == value ? 0 : 2
+
+    number_with_precision(value, precision:, delimiter: ",")
+  end
+
+  def smod_assets?
+    house_in_dispute || vehicle_in_dispute || in_dispute.present?
+  end
+
+  def additional_property_in_dispute?
+    in_dispute.include? "property"
+  end
+
+  def savings_in_dispute?
+    in_dispute.include? "savings"
+  end
+
+  def investments_in_dispute?
+    in_dispute.include? "investments"
+  end
+
+  def valuables_in_dispute?
+    in_dispute.include? "valuables"
+  end
+
+  def main_home_percentage_owned
+    if joint_ownership
+      percentage_owned + joint_percentage_owned
+    else
+      percentage_owned || partner_percentage_owned
+    end
+  end
+
+  def smod_total
+    "TODO"
+  end
+
+  def non_smod_total
+    "TODO"
+  end
+
+  def additional_non_smod_properties_value
+    additional_non_smod_properties_sum "value"
+  end
+
+  def additional_non_smod_properties_mortgage
+    additional_non_smod_properties_sum "outstanding_mortgage"
+  end
+
+  def additional_non_smod_properties_percentage_owned
+    # If there are 2 additional properties and a different percentage of each is owned,
+    # we can't necessarily give a sensible figure here, so leave it blank
+    percentages = additional_non_smod_properties.map { _1["percentage_owned"] }
+    return unless percentages.uniq.length == 1
+
+    percentages.first
+  end
+
+  def additional_non_smod_properties_net_value
+    additional_non_smod_properties_sum "net_value"
+  end
+
+  def additional_non_smod_properties_net_equity
+    additional_non_smod_properties_sum "net_equity"
+  end
+
+  def additional_non_smod_properties_assessed_equity
+    additional_non_smod_properties_sum "assessed_equity"
+  end
+
+  def client_capital_relevant?
+    StepsHelper.valid_step?(session_data, :assets)
+  end
+
+  def client_income_relevant?
+    StepsHelper.valid_step?(session_data, :other_income)
+  end
+
+  def partner_income_relevant?
+    StepsHelper.valid_step?(session_data, :partner_other_income)
+  end
+
+  def not_passporting
+    !passporting
+  end
+
+  def client_non_employment_income
+    non_employment_income
+  end
+
+  def partner_non_employment_income
+    non_employment_income("partner_")
+  end
+
+  def client_mortgage
+    main_home_owned? ? net_housing_costs : 0
```

Comment:
So this one comes from the requirements in the spreadsheet which says: "Any field not populated should contain 0."

It's worth saying that due to the `skip_unless: client_income_relevant?` line in the config file, this zero will only appear if the client _has_ entered housing costs, it's just that we've inferred that these costs are rent rather than mortgage payments. If the client receives passporting benefits or asylum support, this whole section will be moot so `client_income_relevant?` will be false so this field will be skipped and left blank.